vms/platformvm/block/builder/builder.go

```diff
@@ -207,9 +195,33 @@ func (b *builder) setNextBuildBlockTime() {
 	)
 
 	// Wake up when it's time to add/remove the next validator
+	b.nextStakerChangeTimeLock.Lock()
+	b.nextStakerChangeTime = nextStakerChangeTime
+	b.nextStakerChangeTimeLock.Unlock()
 	b.timer.SetTimeoutIn(waitTime)
 }
 
+func (b *builder) maybeAdvanceTime() {
+	ctx := b.txExecutorBackend.Ctx
+
+	// Grabbing the lock here enforces that this function is not called mid-way
+	// through modifying of the state.
+	ctx.Lock.Lock()
+	defer ctx.Lock.Unlock()
+
+	b.nextStakerChangeTimeLock.RLock()
+	defer b.nextStakerChangeTimeLock.RUnlock()
+
+	now := b.txExecutorBackend.Clk.Time()
+	if b.nextStakerChangeTime.After(now) {
+		// [nextStakerChangeTime] is in the future, no need to advance time.
+		return
+	}
```

Comment:
In what situation does this occur?

Reply:
This shouldn't occur under normal operations ever from what I can tell. Added a `ResetBlockTimer` just in case